### PR TITLE
*: only get master status once when query-status

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 
originally we get master status in every sync and relay unit, which is redundant because a worker handles only one source

### What is changed and how it works?
get master status at worker level

### Check List 

Tests 

 - Pass original tests

Code changes

 - Has interface methods change

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: dm/worker/server.go [hunk 1]
```diff
@@ -765,11 +765,13 @@ func (s *Server) QueryStatus(ctx context.Context, req *pb.QueryStatusRequest) (*
...
 	}
 
-	resp.SubTaskStatus, sourceStatus.RelayStatus = w.QueryStatus(ctx, req.Name)
-	unifyMasterBinlogPos(resp, w.cfg.EnableGTID)
+	var err error
+	resp.SubTaskStatus, sourceStatus.RelayStatus, err = w.QueryStatus(ctx, req.Name)
 
 	if len(resp.SubTaskStatus) == 0 {
 		resp.Msg = "no sub task started"
+	} else if err != nil {
```
Comment (glorv) [inline dm/worker/server.go:773]: Should check `err != nil` first

**Review by lichunzhu — COMMENTED** [review]
/lgtm

**Pull request merged**